src/pages/SubmissionDetails.test.js

```diff
@@ -0,0 +1,229 @@
+        run_id: '55b36a53-2ff3-41d0-adc4-abc08aea88ad',
+        engine_id: 'd16721eb-8745-4aa2-b71e-9ade2d6575aa',
+        run_set_id: '0cd15673-7342-4cfa-883d-819660184a16',
+        record_id: 'FOO1',
+        workflow_url: 'https://raw.githubusercontent.com/broadinstitute/cromwell/a40de672c565c4bbd40f57ff96d4ee520dc2b4fc/centaur/src/main/resources/standardTestCases/hello/hello.wdl',
+        state: 'COMPLETE',
+        workflow_params: '[{\'input_name\':\'wf_hello.hello.addressee\',\'input_type\':{\'type\':\'primitive\',\'primitive_type\':\'String\'},\'source\':{\'type\':\'record_lookup\',\'record_attribute\':\'foo_name\'}}]',
+        workflow_outputs: '[]',
+        submission_date: '2022-12-08T23:29:18.675+00:00',
+        last_modified_timestamp: '2022-12-08T23:29:55.695+00:00'
+      }
+    ]
+  }
+
+  const runSetData = {
+    run_sets: [
+      {
+        run_set_id: 'e8347247-4738-4ad1-a591-56c119f93f58',
+        method_id: '00000000-0000-0000-0000-000000000004',
+        is_template: false,
+        run_set_name: 'hello world',
+        run_set_description: 'test',
+        state: 'COMPLETE',
+        record_type: 'FOO',
+        submission_timestamp: '2022-12-08T23:28:50.280+00:00',
+        last_modified_timestamp: '2022-12-09T16:30:50.280+00:00',
+        run_count: 1,
+        error_count: 0,
+        input_definition: '[{\'input_name\':\'wf_hello.hello.addressee\',\'input_type\':{\'type\':\'primitive\',\'primitive_type\':\'String\'},\'source\':{\'type\':\'record_lookup\',\'record_attribute\':\'foo_name\'}}]',
+        output_definition: '[]'
+      }
+    ]
+  }
+
+  const methodData = {
+    methods: [
+      {
+        method_id: '00000000-0000-0000-0000-000000000004',
+        name: 'Hello world',
+        description: 'Add description',
+        source: 'Github',
+        source_url: 'https://raw.githubusercontent.com/broadinstitute/cromwell/a40de672c565c4bbd40f57ff96d4ee520dc2b4fc/centaur/src/main/resources/standardTestCases/hello/hello.wdl',
+        created: '2022-12-08T23:28:50.280+00:00'
+      }
+    ]
+  }
+
+  const submissionId = 'e8347247-4738-4ad1-a591-56c119f93f58'
+
+  beforeAll(() => {
+    Object.defineProperty(HTMLElement.prototype, 'offsetHeight', { configurable: true, value: 1000 })
+    Object.defineProperty(HTMLElement.prototype, 'offsetWidth', { configurable: true, value: 800 })
+  })
+
+  beforeEach(() => {
+    const getRunsMethod = jest.fn(() => Promise.resolve(runsData))
+    Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runs: {
+            get: getRunsMethod
+          }
+        }
+      }
+    })
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  afterAll(() => {
+    Object.defineProperty(HTMLElement.prototype, 'offsetHeight', originalOffsetHeight)
+    Object.defineProperty(HTMLElement.prototype, 'offsetWidth', originalOffsetWidth)
+  })
+
+  it('should correctly display previous 2 runs', async () => {
+    // Act
+    await act(async () => {
```

Comment:
note: there's no need to change anything here. I'm just passing on some info I learned the other day from Miguel.

apparently, it's a "best practice" not to use `act` unless you really need it. I don't know what situation calls for it or not -- if you felt like it, you could see if this test works the same way if you replace these three lines with just `render(h(SubmissionDetails))` (without the wrapping `act` and without `await`s)

Reply:
Oh, thank you. I'll test this out and see if anything changes.

Reply:
I was able to change this in one place, but it seems that data wasn't able to be rendered when omitting the `act` from tests that were testing table functions/features.